src/main/java/frc/robot/lib/FieldGeometry.kt

```diff
@@ -0,0 +1,59 @@
+package frc.robot.lib
+
+import edu.wpi.first.math.geometry.Rotation2d
+import edu.wpi.first.math.geometry.Translation2d
+import edu.wpi.first.wpilibj.DriverStation
+import frc.robot.Robot
+
+/** Object for defining field geometry and functions to measure distances */
+object FieldGeometry {
+    const val FIELD_X_LENGTH = 17.530
+    const val FIELD_Y_LENGTH = 8.230
+
+    // These are constructed so that left hand perpendicular is always pointing towards the center
+    // of the field
+    private val BLUE_CORAL_STATION_LOWER =
+        LineSegment(Translation2d(0.398, 0.985), Translation2d(1.36, 0.291))
+    private val BLUE_CORAL_STATION_UPPER =
+        LineSegment(Translation2d(1.373, 7.757), Translation2d(0.409, 7.065))
+    private val RED_CORAL_STATION_LOWER =
+        LineSegment(BLUE_CORAL_STATION_LOWER.p2.mirror(), BLUE_CORAL_STATION_LOWER.p1.mirror())
+    private val RED_CORAL_STATION_UPPER =
+        LineSegment(BLUE_CORAL_STATION_UPPER.p2.mirror(), BLUE_CORAL_STATION_UPPER.p1.mirror())
+
+    private val CORAL_STATIONS
+        get() =
+            if (Robot.alliance == DriverStation.Alliance.Blue) {
+                listOf(BLUE_CORAL_STATION_LOWER, BLUE_CORAL_STATION_UPPER)
+            } else {
+                listOf(RED_CORAL_STATION_LOWER, RED_CORAL_STATION_UPPER)
+            }
+
+    private val BLUE_BARGE_ALIGNMENT_LINE =
```

Comment:
We can align to the barge from both sides, so we should add the alignment from the other side as well